### PR TITLE
fix(navigation): Simplify initial route logic

### PR DESCRIPTION
**Pull request by jophish**
### Description

This PR overhauls how we handle the initial route logic for the app. The main difficulty in determining the initial route is onboarding. Previously, we collected various bits and pieces of random information and used them in aggregate, along with some custom logic, as a proxy for what screen the user left off at during onboarding. With this PR, we write state directly to redux during onboarding about the _first screen_ present in the user's current onboarding step. On startup, we set the initial route to this screen, less a couple of edge cases. (This means that a user will always be dropped back at the _beginning_ of their current onboarding step.)

### Test plan

Unit and manual tested. See video below.

https://github.com/user-attachments/assets/a2312a02-a590-483e-953d-ebcebc25c509

### Related issues

- Fixes #[issue number here]

### Network scalability

If a new NetworkId and/or Network are added in the future, the changes in this PR will:

- [ ] Continue to work without code changes, OR trigger a compilation error (guaranteeing we find it when a new network is added)

### CODE REVIEW

File: src/identity/reducer.ts [hunk 1]
```diff
@@ -55,7 +55,6 @@ export interface AddressToVerificationStatus {
 }
 
 interface State {
-  hasSeenVerificationNux: boolean
```
Comment (jophish) [inline src/identity/reducer.ts:58]: This bit of state is fine to get rid of, since the only place we used it was the initial route logic anyways.

File: src/navigator/Navigator.tsx [hunk 2]
```diff
@@ -473,6 +474,11 @@ const settingsScreens = (Navigator: typeof Stack) => (
       name={Screens.SignInWithEmail}
       options={noHeader}
       component={SignInWithEmail}
+      initialParams={{
```
Comment (jophish) [inline src/navigator/Navigator.tsx:477]: As far as I'm aware, with react-navigation v6, there is no way to pass a set of "initial params" to the initial route on the navigator itself. This was not the case in earlier versions. Instead, you must provide an `initialParams` object to screens themselves, which, as far as I'm aware, will be used in place of params on the first call if none are present.

Only one screen during onboarding actually has props, the `SignInWithEmail` screen. In my first pass at this, I stored these params in Redux, but since there's no easy way to set these initial props on screens dynamically, I opted for this approach instead. It's kind of lame, having this one special case, but I think it's probably OK.

Another thing to note here, the `SignInWithEmail` screen has a back button, which, if it's the initial route, doesn't do anything. This (and the corresponding changes within the screen itself) cause the back button to be hidden if the screen is being rendered as the initial route. It's not really the end of the world if the back button is there and doesn't do anything, but I thought it would be cleaner to hide it.

File: src/onboarding/steps.ts [hunk 3]
```diff
@@ -215,11 +217,16 @@ export function _getStepInfo({ firstScreenInStep, navigator, dispatch, props }:
     skipProtectWallet,
   } = props
 
+  function wrapNavigate<T extends keyof StackParamList>(screen: T, props: StackParamList[T]) {
```
Comment (satish-ravi) [inline src/onboarding/steps.ts:220]: you can use the same type as the type of navigate from NavigationService (maybe they can be shared?) That way you can avoid the extra `undefined` in the calls below

File: src/redux/migrations.ts [hunk 4]
```diff
@@ -1925,4 +1931,47 @@ export const migrations = {
...
   235: (state: any) => state,
+  236: (state: any) => {
+    // This migration is taken directly from the old initial route logic. The
+    // only difference is that the multi-chain beta screen has been removed, since
+    // it required fetching a feature gate.
+    const lastOnboardingStepScreen = (() => {
+      if (!state.i18n.language) {
+        return Screens.Language
+      } else if (!state.account.acceptedTerms || state.account.pincodeType === PincodeType.Unset) {
+        // allow empty username
```
Comment (satish-ravi) [inline src/redux/migrations.ts:1942]: this comment seems outdated

File: src/keylessBackup/SignInWithEmail.tsx [hunk 5]
```diff
@@ -196,15 +196,17 @@ function SignInWithEmail({ route }: Props) {
...
+          !hideBack ? (
             <BackButton
               eventName={KeylessBackupEvents.cab_sign_in_with_email_screen_cancel}
               eventProperties={{
                 keylessBackupFlow,
                 origin,
               }}
             />
+          ) : (
+            <></>
```
Comment (satish-ravi) [inline src/keylessBackup/SignInWithEmail.tsx:209]: this can just be undefined instead of an empty view right?

File: src/navigator/Navigator.tsx (hunk 2 above)
Comment (satish-ravi) [reply src/navigator/Navigator.tsx:477]: Is there a way we can check if back is possible? would be nice to avoid the `hideBack` option if so. Also I think @MuckT is doing an upgrade to v7, wonder if v7 has a better way of handling this?

**Review by satish-ravi — COMMENTED** [review]
Can you update the test plan? Seems like this could use some manual tests

File: src/navigator/Navigator.tsx (hunk 2 above)
Comment (jophish) [reply src/navigator/Navigator.tsx:477]: There's a [canGoBack](https://reactnavigation.org/docs/6.x/navigation-prop/#cangoback) method in v6, which the documentation states should not be used for rendering purposes since it doesn't trigger a re-render. I think in this case the lack of re-rendering is probably fine, right, since there would _never_ be a way to go back from the initial screen.

File: src/onboarding/steps.ts (hunk 3 above)
Comment (jophish) [reply src/onboarding/steps.ts:220]: Ah yeah, I had tried this but encountered a strange typing problem. I think I've gotten it to work now, however

File: src/navigator/Navigator.tsx (hunk 2 above)
Comment (MuckT) [reply src/navigator/Navigator.tsx:477]: Looks like [canGoBack](https://reactnavigation.org/docs/navigation-object/#cangoback) is supported in v7.

File: src/identity/reducer.ts (hunk 1 above)
Comment (jh2oman) [reply src/identity/reducer.ts:58]: 🙏

**Review by jh2oman — APPROVED** [review]
Awesome simplification. Thanks joe!
